src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Models/KeyLabelIdentifier.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Models
+{
+    internal struct KeyLabelIdentifier
```

Comment:
>KeyLabelIdentifier [](start = 20, length = 18)

Since we now have a separate dictionary to store watched settings, is it possible for us to reuse the `KeyValueWatcher` reference as the dictionary key and avoid introducing a new type?

Reply:
Since we need something to serve as dictionary key, I think `KeyLabel` helps us create a smaller object every time we do a dictionary lookup. `KeyValueWatcher` has a lot of other fields that we don't need for this scenario.

Reply:
We'll stick with KeyValueIdentifier for now and we can discuss it again when you are back.